benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp

```diff
@@ -586,14 +600,79 @@ void GraphicsBenchmarkApp::ProcessQuadsKnobs()
     }
 }
 
+#if defined(PPX_BUILD_XR)
+void GraphicsBenchmarkApp::RecordAndSubmitCommandBufferGUIXR(PerFrame& frame)
+{
+    uint32_t           imageIndex  = UINT32_MAX;
+    grfx::SwapchainPtr uiSwapchain = GetUISwapchain();
+    PPX_CHECKED_CALL(uiSwapchain->AcquireNextImage(UINT64_MAX, nullptr, nullptr, &imageIndex));
+    PPX_CHECKED_CALL(frame.uiRenderCompleteFence->WaitAndReset());
+
+    PPX_CHECKED_CALL(frame.uiCmd->Begin());
+    {
+        grfx::RenderPassPtr renderPass = uiSwapchain->GetRenderPass(imageIndex, grfx::ATTACHMENT_LOAD_OP_CLEAR);
+        PPX_ASSERT_MSG(!renderPass.IsNull(), "render pass object is null");
+
+        frame.uiCmd->BeginRenderPass(renderPass);
+        // Draw ImGui
+        UpdateGUI();
+        DrawImGui(frame.uiCmd);
+        frame.uiCmd->EndRenderPass();
+    }
+    PPX_CHECKED_CALL(frame.uiCmd->End());
+
+    grfx::SubmitInfo submitInfo     = {};
+    submitInfo.commandBufferCount   = 1;
+    submitInfo.ppCommandBuffers     = &frame.uiCmd;
+    submitInfo.waitSemaphoreCount   = 0;
+    submitInfo.ppWaitSemaphores     = nullptr;
+    submitInfo.signalSemaphoreCount = 0;
+    submitInfo.ppSignalSemaphores   = nullptr;
+
+    submitInfo.pFence = frame.uiRenderCompleteFence;
+
+    PPX_CHECKED_CALL(GetGraphicsQueue()->Submit(&submitInfo));
+}
+#endif
+
 void GraphicsBenchmarkApp::Render()
```

Comment:
This was a bit hard to read with 5 blocks of `#if defined(PPX_BUILD_XR)`, I wonder if it would just be better if there was a separate `RenderXR` function for the case of (`PPX_BUILD_XR==1` and `IsXrEnabled()==true`) ?

Reply:
i would prefer to avoid duplicating the code so that in the future when people changes the Render() function, they don't need to do the change in 2 places. Although I do agree that the `#if defined(PPX_BUILD_XR)` is annoying. I am planning to do another pass to move all `#if defined(PPX_BUILD_XR)` into the base class or framework so that we don't do this ifdefs in the application.